external-crates/move/documentation/book/src/unit-testing.md

```diff
@@ -43,96 +44,235 @@ public fun this_test_will_abort_and_pass() { abort 1 }
 #[expected_failure]
 public fun test_will_error_and_pass() { 1/0; }
 
-#[test]
-#[expected_failure(abort_code = 0)]
+#[test] // Will pass since test fails with the expected abort code constant.
+#[expected_failure(abort_code = ENotFound)] // ENotFound is a constant defined in the module
+public fun test_will_error_and_pass_abort_code() { abort ENotFound }
+
+#[test] // Will fail since test fails with a different error than expected.
+#[expected_failure(abort_code = my_module::EnotFound)]
 public fun test_will_error_and_fail() { 1/0; }
 
 #[test, expected_failure] // Can have multiple in one attribute. This test will pass.
 public fun this_other_test_will_abort_and_pass() { abort 1 }
 ```
 
-With arguments, a test annotation takes the form
-`#[test(<param_name_1> = <address>, ..., <param_name_n> = <address>)]`. If a function is annotated
-in such a manner, the function's parameters must be a permutation of the parameters
-<`param_name_1>, ..., <param_name_n>`, i.e., the order of these parameters as they occur in the
-function and their order in the test annotation do not have to be the same, but they must be able to
-be matched up with each other by name.
+## Expected Failures
+
+There are a number of different ways that you can use the `#[expected_failure]`
+annotation to specify different types of error conditions. These are:
 
-Only parameters with a type of `signer` are supported as test parameters. If a non-`signer`
-parameter is supplied, the test will result in an error when run.
+### 1. `#[expected_failure]` 
 
+This will pass if the test aborts with any error code. Examples of these types of annotations are:
+
+```move
+#[test]
+#[expected_failure]
+fun test_will_abort_and_pass1() { abort 1 }
+
+#[test]
+#[expected_failure]
+fun test_will_arith_error_and_pass2() { 1/0; }
 ```
-#[test(arg = @0xC0FFEE)] // OK
-fun this_is_correct_now(arg: signer) { ... }
 
-#[test(wrong_arg_name = @0xC0FFEE)] // Not correct: arg name doesn't match
-fun this_is_incorrect(arg: signer) { ... }
+### 2. `#[expected_failure(abort_code = <constant>)]`
+
+This will pass if the test
+aborts with the specified constant value in the module that defines the
+constant, and fail otherwise.
```

Comment:
We should highlight immediately that this is the recommended way for most tests. And maybe even reorder these items to stress that?